dbms/src/Dictionaries/CacheDictionary.cpp

```diff
@@ -832,9 +835,14 @@ void CacheDictionary::update(BunchUpdateUnit & bunch_update_unit) const
             }
 
             Stopwatch watch;
-            auto stream = source_ptr->loadIds(bunch_update_unit.getRequestedIds());
 
-            const ProfilingScopedWriteRWLock write_lock{rw_lock, ProfileEvents::DictCacheLockWriteNs};
+            /// To perform parallel loading.
+            BlockInputStreamPtr stream = nullptr;
+            {
+                write_lock.unlock();
+                stream = source_ptr->loadIds(bunch_update_unit.getRequestedIds());
+                write_lock.lock();
+            }
 
             stream->readPrefix();
             while (const auto block = stream->read())
```

Comment:
read() might block for some storages, so maybe we should release the lock here as well.